Shared/Extensions/String+Extension.swift

```diff
@@ -131,6 +131,20 @@ public extension String {
         return nsString.substring(with: match.range(at: 1))
     }
 
+    /// Calculates and returns the width of a string when rendered with a specified font.
+    ///
+    /// - Parameter font: The font that will be used to render the string.
+    ///
+    /// - Returns: The width of the string when rendered with the specified font.
+    func width(with font: UIFont) -> CGFloat {
```

Comment:
IMO I would keep this in the FakeSpot VC. I don't think we want this to be a standard practice to measure the width of a string. It should be an exception

Reply:
I get where Sorin's coming from. It's kind of a Swift thing to do, and sticking it in Fakespot VC could clutter things up with extras that aren't really about what the VC is supposed to do. Extensions are pretty handy in Swift for this sort of thing, so maybe that's the way to go.

Reply:
It's a basic Swift thing to do yes and most people do it by default, but we challenged that in previous eng sync meetings. We also had this conversation multiple times together already, but as said in the past we're trying to avoid extension usage unless specifically needed. Measuring the width of a string isn't a behavior we should have by default as it encourage going away from supporting automatic sizing, which is needed for dynamic type. It can be done where needed, but I wouldn't vote to encourage such behavior in our app. Happy to discuss further if needed and bring this up to the team.